test-engineering/load-tests/setup_k8.sh

```diff
@@ -0,0 +1,110 @@
+
+LOCUST_IMAGE_TAG=$(git log -1 --pretty=format:%h)
+echo "Docker image tag for locust is set to: ${LOCUST_IMAGE_TAG}"
+
+##Kubernetes Manifests variables
+KINTO__SERVER_URL='https://firefox.settings.services.mozilla.com'
+KINTO__COLLECTION=quicksuggest
+KINTO__BUCKET=main
+
+SetupGksCluster()
+{
+
+    #Configure Kubernetes
+    echo -e "==================== Prepare environments with set of environment variables "
+    echo -e "==================== Set Kubernetes Cluster "
+    export CLUSTER=$CLUSTER
+    echo -e "==================== Set Kubernetes TARGET "
+    export TARGET=$TARGET
+    echo -e "==================== Set SCOPE "
+    export SCOPE=$SCOPE
+
+    echo -e "==================== Refresh Kubeconfig at path ~/.kube/config "
+    $GCLOUD container clusters get-credentials $CLUSTER --region $REGION --project $GOOGLE_CLOUD_PROJECT
+
+    ##Build Docker Images
+    echo -e "==================== Build the Docker image and store it in your project's container registry. Tag with the latest commit hash "
+    $GCLOUD builds submit --tag gcr.io/$GOOGLE_CLOUD_PROJECT/locust-merino:$LOCUST_IMAGE_TAG
+    echo -e "==================== Verify that the Docker image is in your project's container repository"
+    $GCLOUD container images list | grep locust-merino
+
+    ##Deploying the Locust master and worker nodes
+    echo -e "==================== Update Kubernetes Manifests "
+    echo -e "==================== Replace the target host and project ID with the deployed endpoint and project ID in the locust-master-controller.yml and locust-worker-controller.yml files"
+
+    FILES=($MASTER_FILE $WORKER_FILE)
+    for file in "${FILES[@]}"
+    do
+        $SED -i -e "s|\[TARGET_HOST\]|$TARGET|g" $MERINO_DIRECTORY/$file
+        $SED -i -e "s|\[PROJECT_ID\]|$GOOGLE_CLOUD_PROJECT|g" $MERINO_DIRECTORY/$file
+        $SED -i -e "s|\[LOCUST_IMAGE_TAG\]|$LOCUST_IMAGE_TAG|g" $MERINO_DIRECTORY/$file
+        $SED -i -e "s|\[KINTO__BUCKET\]|$KINTO__BUCKET|g" $MERINO_DIRECTORY/$file
+        $SED -i -e "s|\[KINTO__COLLECTION\]|$KINTO__COLLECTION|g" $MERINO_DIRECTORY/$file
+        $SED -i -e "s|\[KINTO__SERVER_URL\]|$KINTO__SERVER_URL|g" $MERINO_DIRECTORY/$file
+        $SED -i -e "s|\[WORKER_COUNT\]|$WORKER_COUNT|g" $MERINO_DIRECTORY/$file
+
+    done
+
+    ##Deploy the Locust master and worker nodes using Kubernetes Manifests
+    echo -e "==================== Deploy the Locust master and worker nodes"
+    $KUBECTL apply -f $MERINO_DIRECTORY/$MASTER_FILE
+    $KUBECTL apply -f $MERINO_DIRECTORY/$SERVICE_FILE
+    $KUBECTL apply -f $MERINO_DIRECTORY/$WORKER_FILE
+
+    echo -e "==================== Verify the Locust deployments & Services"
+    $KUBECTL get pods -o wide
+    $KUBECTL get services
+}
+
+echo "==================== The script is used to create & delete the GKE cluster"
+echo "==================== Do you want to create or setup the existing or delete GKE cluster? Select ${BOLD}create or delete or setup ${NORM}"
+while :
+do
```

Comment:
Sorry, I'm not good at Bash scripting, but does this infinite loop until you hit Ctrl+C or will this exit the script as soon as you type in "create<enter>"?

Trying to figure out if I can run <kbd>./setup_k8.sh</kbd> and still run <kbd>kubectl get pods -o wide</kbd> in the same terminal or get the external IP.

Reply:
once you type 'create' it performs all the step until getting the service started and then you can run any commands without hitting Ctrl+C. To get the external IP we any ways have to wait or run the watch command in the terminal.